src/coreclr/src/System.Private.CoreLib/src/System/StubHelpers.cs

```diff
@@ -686,11 +686,61 @@ internal static long ConvertToManaged(double nativeDate)
 #if FEATURE_COMINTEROP
     internal static class InterfaceMarshaler
     {
+        // See interopconverter.h
+        [Flags]
+        private enum ItfMarshalFlags : int
+        {
+            ITF_MARSHAL_INSP_ITF        = 0x01,
+            ITF_MARSHAL_SUPPRESS_ADDREF = 0x02,
+            ITF_MARSHAL_CLASS_IS_HINT   = 0x04,
+            ITF_MARSHAL_DISP_ITF        = 0x08,
+            ITF_MARSHAL_USE_BASIC_ITF   = 0x10,
+            ITF_MARSHAL_WINRT_SCENARIO  = 0x20,
+        };
+
+        private static bool IsForIUnknown(int flags)
+        {
+            ItfMarshalFlags interfaceFlags = (ItfMarshalFlags)flags;
+            return (interfaceFlags & ItfMarshalFlags.ITF_MARSHAL_USE_BASIC_ITF) != 0
```

Comment:
With the `Flags` attribute can't we use the [`HasFlag()`](https://docs.microsoft.com/dotnet/api/system.enum.hasflag)?

Reply:
We could, but it is smaller and faster code to do it as bit operation.

Reply:
Oh nevermind then. I thought the `HasFlag()` represented a heavily optimized path.